### PR TITLE
Introduce `Instant{Identity,TruncatedTo{Milliseconds,Seconds}}` Refaster rules

### PR DESCRIPTION
**Pull request by werli**
Some trivial rules I noticed when drafting up other rules I needed to omit.

### CODE REVIEW

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/TimeRules.java [hunk 1]
```diff
@@ -142,6 +142,24 @@ OffsetDateTime after(Instant instant, ZoneId zoneId) {
...
+  /** Don't unnecessarily transform an {@link Instant} to an equivalent instance. */
+  static final class InstantIdentity {
+    @BeforeTemplate
+    Instant before(Instant instant) {
+      return Refaster.anyOf(
+          instant.plus(Duration.ZERO),
+          instant.minus(Duration.ZERO),
+          Instant.parse(instant.toString()),
+          Instant.ofEpochMilli(instant.toEpochMilli()),
+          Instant.ofEpochSecond(instant.getEpochSecond()));
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/TimeRules.java]: These two aren't actually identity transformations, as they drop the nano and millisecond components, respectively. Let's introduce rules to suggest using `Instant#truncatedTo` instead.

**Review by Stephan202 — APPROVED** [review]
Rebased and added a commit. Suggested commit message:
```
Introduce `Instant{Identity,TruncatedTo{Milliseconds,Seconds}}` Refaster rules (#1395)

While there, have `IdentityConversion` support `Instant#from`.

**Review by rickie — APPROVED** [review]
Nice cleanup rules :rocket:!

**Comment by rickie** [comment]
@Venorcis anything from your side? :)

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/TimeRules.java (hunk 1 above)
Comment (werli) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/TimeRules.java]: Makes sense, thanks 💪

**Review by Venorcis — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**